schema/validator.go

```diff
@@ -26,14 +26,42 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
+type validateRefMediaFunc func(r io.Reader) error
+
+var vlidateRefMediaMap = map[Validator]validateRefMediaFunc{
+	ValidatorMediaTypeImageIndex: validateImageIndexRefObject,
+	ValidatorMediaTypeManifest:   validateManifestRefObject,
+}
+
+// ValidateFunc provides the type of function to validate specific object.
+type ValidateFunc func(r io.Reader, v Validator) error
+
 // Validator wraps a media type string identifier
 // and implements validation against a JSON schema.
 type Validator string
 
-type validateDescendantsFunc func(r io.Reader) error
+// Validate validates the given reader against the schema and OCI format defined in spec.
```

Comment:
I don't see anything about “the schema” or “OCI format defined in the spec” in your current implementation (3ad5d1b).  This function is just “read `r` into memory and call all of `funcList` on the content, dying on the first error”.